cookiecutter/cli.py

```diff
@@ -67,4 +68,9 @@ def main(template, no_input, checkout, verbose):
             level=logging.INFO
         )
 
-    cookiecutter(template, checkout, no_input)
+    if extra_context:
+        # Convert tuple -- e.g.: (u'program_name=foobar',)
+        # to dict -- e.g.: {'program_name': 'foobar'}
+        extra_context = dict(s.split('=', 1) for s in extra_context)
```

Comment:
Can we please add some validation here, this mini parser is a little brittle if it doesn't encounter the happy case.